File: control-room/SosRelay/EventTypes/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SOS.EventTypes
{
    public class CarLocation : Location, Orientation
    {
        public double X { get; set; } = 0;
        public double Y { get; set; } = 0;
        public double Z { get; set; } = 0;
        public int pitch { get; set; } = 0;
        public int roll { get; set; } = 0;
        public int yaw { get; set; } = 0;
    }

    public class Player
    {
        public int assists { get; set; } = 0;
        public string attacker { get; set; } = string.Empty;
        public int boost { get; set; } = 0;
        public int cartouches { get; set; } = 0;
        public int demos { get; set; } = 0;
        public int goals { get; set; } = 0;
        public bool hasCar { get; set; } = false;
        public string id { get; set; } = string.Empty;
        public bool isDead { get; set; } = false;
        public bool isPowersliding { get; set; } = false;
        public bool isSonic { get; set; } = false;
        public CarLocation location { get; set; } = new CarLocation();
        public string name { get; set; } = string.Empty;
        public bool onGround { get; set; } = false;
        public bool onWall { get; set; } = false;
        public string primaryID { get; set; } = string.Empty;
        public int saves { get; set; } = 0;
        public int score { get; set; } = 0;
        public int shortcut { get; set; } = 0;
        public int shots { get; set; } = 0;
        public int speed { get; set; } = 0;
        public int team { get; set; } = 0;
        public int touches { get; set; } = 0;
    }
}

File: overlay-app/src/types/player.ts
export interface PlayerLocation {
  X: number;
  Y: number;
  Z: number;
  pitch: number;
  roll: number;
  yaw: number;
}

export interface Player {
  assists: number;
  attacker: string;
  boost: number;
  cartouches: number;
  demos: number;
  goals: number;
  hasCar: boolean;
  id: string;
  isDead: boolean;
  isPowersliding: boolean;
  isSonic: boolean;
  location: {
    X: number;
    Y: number;
    Z: number;
    pitch: number;
    roll: number;
    yaw: number;
  };
  name: string;
  onGround: boolean;
  onWall: boolean;
  primaryID: string;
  saves: number;
  score: number;
  shortcut: number;
  shots: number;
  speed: number;
  team: number;
  touches: number;
}

export function NewPlayer(): Player {
  return {
    name: '',
    score: 0,
    goals: 0,
    assists: 0,
    shots: 0,
    saves: 0,
    demos: 0,
    attacker: '',
    boost: 0,
    cartouches: 0,
    hasCar: false,
    id: '',
    isDead: false,
    isPowersliding: false,
    isSonic: false,
    location: {
      X: 0,
      Y: 0,
      Z: 0,
      pitch: 0,
      roll: 0,
      yaw: 0,
    },
    onGround: false,
    onWall: false,
    primaryID: '',
    shortcut: 0,
    speed: 0,
    team: 0,
    touches: 0,
  };
}
